docs/reference/extensions.rst

```diff
@@ -0,0 +1,16 @@
+Extensions
+**********
+
+Just as the snapcraft extensions are designed to simplify snap creation,
+Rockcraft extensions are crafted to expand and modify the user-provided
+rockcraft manifest file, aiming to minimise the boilerplate code when
```

Comment:
hmm not sure we should use the word "manifest" here, I don't think we use it anywhere else in the docs. "project" is probably more appropriate.